images/syncer/Dockerfile

```diff
@@ -17,7 +17,7 @@
 ARG GOLANG_IMAGE=golang:1.21
 
 # This build arg allows the specification of a custom base image.
-ARG BASE_IMAGE=gcr.io/cloud-provider-vsphere/extra/csi-driver-base:latest
+ARG BASE_IMAGE=photon:4.0
```

Comment:
Is there a possibility of observing "docker image pull rate limit" error from `docker.io/library/photon:4.0` ?

Reply:
Good question @chethanv28, the limit is 100 requests per IP per 6 hours. Depending on how the runners are setup the image may be cached between runs.

Reply:
We need to make sure our internal pipelines are not affected due to this rate limit.

cc: @sashrith

Reply:
we are setting this base image variable while building the syncer and driver images which is passed to docker buildx to override the default in the dockerfile
https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/334523664631a6a6d650cb1fa087d2d9e3f8b5d8/hack/release.sh#L60
we can do something similar to that here as well

Reply:
I can change that @sashrith, but is all of CI using the `release.sh` script? If not then we may not want to pass in the ARG since now we would need to remember to change the value in multiple different places.

I also see https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/334523664631a6a6d650cb1fa087d2d9e3f8b5d8/.gitlab-ci.yml#L40-L41, is this still being used in CI?